provider/agent.go

```diff
@@ -272,6 +260,23 @@ func agentResource() *schema.Resource {
 				Optional:    true,
 			},
 		},
+		CustomizeDiff: func(ctx context.Context, rd *schema.ResourceDiff, i interface{}) error {
+			if !rd.HasChange("metadata") {
+				return nil
+			}
+
+			keys := map[string]struct{}{}
+			for _, t := range rd.Get("metadata").([]interface{}) {
```

Comment:
Would it make sense to check that `rd.Get("metadata")` can be cast to a `[]interface{}`?
Same for below with `taint`.

Also, I think you can just use `any` as an alias for `interface{}` now.

Reply:
I don't think it's possible for either cast to fail, but I'll add checks in-case the schema changes